diff --git a/arch/arm64/boot/dts/amlogic/mesong12_odroid_common.dtsi b/arch/arm64/boot/dts/amlogic/mesong12_odroid_common.dtsi
old mode 100644
new mode 100755
index 54b4075f9db0..1f2b6719bb6f
--- a/arch/arm64/boot/dts/amlogic/mesong12_odroid_common.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesong12_odroid_common.dtsi
@@ -60,11 +60,16 @@
 	};
 
 	/* Audio Related start */
-	dummy_codec:dummy{
+	spdif_dit:spdif{
 		#sound-dai-cells = <0>;
 		compatible = "linux,spdif-dit";
 		status = "okay";
 	};
+	dummy_codec:dummy{
+		#sound-dai-cells = <0>;
+		compatible = "amlogic, aml_dummy_codec";
+		status = "okay";
+	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
@@ -106,7 +111,7 @@
 				system-clock-frequency = <6144000>;
 			};
 			codec {
-				sound-dai = <&dummy_codec>;
+				sound-dai = <&spdif_dit>;
 			};
 		};
 
@@ -115,7 +120,6 @@
 			mclk-fs = <256>;
 			bitclock-master = <&aml_tdmc>;
 			frame-master = <&aml_tdmc>;
-			continuous-clock;
 			cpu {
 				sound-dai = <&aml_tdmc>;
 				dai-tdm-slot-tx-mask = <1 1>;
@@ -137,6 +141,24 @@
 				sound-dai = <&aml_spdif>;
 				system-clock-frequency = <6144000>;
 			};
+			codec {
+				sound-dai = <&spdif_dit>;
+			};
+		};
+
+		aml-audio-card,dai-link@3 {
+			format = "i2s";
+			mclk-fs = <256>;
+			bitclock-master = <&aml_tdmb>;
+			frame-master = <&aml_tdmb>;
+			cpu {
+				sound-dai = <&aml_tdmb>;
+				dai-tdm-slot-tx-mask = <1 1>;
+				dai-tdm-slot-rx-mask = <1 1>;
+				dai-tdm-slot-num = <2>;
+				dai-tdm-slot-width = <32>;
+				system-clock-frequency = <12288000>;
+			};
 			codec {
 				sound-dai = <&dummy_codec>;
 			};
@@ -381,6 +403,21 @@
 };
 
 &audiobus {
+	aml_tdmb: tdmb {
+		compatible = "amlogic, g12a-snd-tdmb";
+		#sound-dai-cells = <0>;
+		dai-tdm-lane-slot-mask-in = <0 1 0 0>;
+		dai-tdm-lane-slot-mask-out = <1 0 0 0>;
+		dai-tdm-clk-sel = <1>;
+		clocks = <&clkaudio CLKID_AUDIO_MCLK_B
+				&clkc CLKID_MPLL1>;
+		clock-names = "mclk", "clk_srcpll";
+		pinctrl-names = "tdm_pins";
+		pinctrl-0 = <&tdmb_mclk &tdmout_b /*&tdmin_b*/>;
+		mclk_pad = <0>;  /* 0: mclk_0; 1: mclk_1 */
+		i2s2hdmi = <0>;
+		status = "okay";
+	};
 	/* tdmc to internal DAC output, no pinmux */
 	aml_tdmc: tdmc {
 		compatible = "amlogic, g12a-snd-tdmc";
@@ -391,6 +428,7 @@
 		clocks = <&clkaudio CLKID_AUDIO_MCLK_C
 				&clkc CLKID_MPLL2>;
 		clock-names = "mclk", "clk_srcpll";
+		mclk_pad = <1>;  /* 0: mclk_0; 1: mclk_1 */
 		i2s2hdmi = <0>;
 		status = "okay";
 	};
@@ -505,6 +543,22 @@
 };
 
 &pinctrl_periphs {
+	tdmb_mclk: tdmb_mclk {
+		mux {
+			groups = "mclk0_a";
+			function = "mclk0";
+			drive-strength = <2>;
+		};
+	};
+	tdmout_b: tdmout_b {
+		mux { /* GPIOA_1, GPIOA_2, GPIOA_3 */
+			groups = "tdmb_sclk",
+				"tdmb_fs",
+				"tdmb_dout0";
+			function = "tdmb_out";
+			drive-strength = <2>;
+		};
+	};
 	i2c2_to_gpiox:i2c2_gpiox {
 		mux {
 			groups = "GPIOX_17",
@@ -527,8 +581,8 @@
 		};
 	};
 	spdifout: spdifout {
-		mux {/* GPIOA_13 */
-			groups = "spdif_out_a13";
+		mux {/* GPIOA_11 */
+			groups = "spdif_out_a11";
 			function = "spdif_out";
 		};
 	};
